src/bootstrap/tool.rs

```diff
@@ -727,6 +734,63 @@ impl Step for RustAnalyzerProcMacroSrv {
     }
 }
 
+#[derive(Debug, Copy, Clone, Hash, PartialEq, Eq, Ord, PartialOrd)]
+pub struct RunMakeSupport {
+    pub compiler: Compiler,
+    pub target: TargetSelection,
+}
+
+impl Step for RunMakeSupport {
+    type Output = PathBuf;
+    const DEFAULT: bool = true;
+
+    fn should_run(run: ShouldRun<'_>) -> ShouldRun<'_> {
+        run.path("src/tools/run-make-support")
+    }
+
+    fn make_run(run: RunConfig<'_>) {
+        run.builder.ensure(RunMakeSupport {
+            compiler: run.builder.compiler(1, run.builder.config.build),
+            target: run.target,
+        });
+    }
+
+    fn run(self, builder: &Builder<'_>) -> PathBuf {
+        let cargo = prepare_tool_cargo(
+            builder,
+            self.compiler,
+            Mode::ToolStd,
+            self.target,
+            "build",
+            "src/tools/run-make-support",
+            SourceType::InTree,
+            &[],
+        );
+        let mut cargo = Command::from(cargo);
+        cargo.env("RUSTFLAGS", "-Cprefer-dynamic");
+        let is_expected = builder.try_run(&mut cargo).is_ok();
+
+        builder.save_toolstate(
+            "src/tools/run-make-support",
+            if is_expected { ToolState::TestFail } else { ToolState::BuildFail },
+        );
+
+        let lib_name = if self.target.contains("windows") {
+            "librun_make_support.dll"
+        } else if self.target.contains("apple-darwin") {
```

Comment:
For iOS this should also use `.dylib` as extension. Looking for `apple` should work I think.